### PR TITLE
Dockerfile refactoring

### PR DESCRIPTION
**Pull request by shleeable**
While this offers no additional features.. the Dockerfile has simply been reorganised for readability and easier changes in the future. Pure Housekeeping.

The code was just shifted around.

Status: Do not test yet.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,117 +1,155 @@
...
-	cd jemalloc-$JE_VER && \
-	./autogen.sh && \
-	./configure --prefix=/opt/jemalloc && \
-	make -j$(nproc) > /dev/null && \
-	make install_bin install_include install_lib
-
-# Install ruby
-ENV RUBY_VER="2.6.5"
+RUN apt-get update && apt-get --no-install-recommends -y install make autoconf gcc g++ && \
+    cd ~ && \
```
Comment (SuperSandro2000) [inline Dockerfile:35]: is ``~`` ``/root/`` or ``/home/mastodon``?

File: Dockerfile [hunk 2]
```diff
@@ -1,117 +1,155 @@
...
 
+##-------------------------------------------------
 FROM ubuntu:18.04
 
+# Use bash for the SHELL
+SHELL ["/bin/bash", "-o", "pipefail", "-c"]
+
+RUN echo "Etc/UTC" > /etc/localtime && \
+    apt-get update && apt-get --no-install-recommends -y install \
+    apt-utils \
```
Comment (SuperSandro2000) [inline Dockerfile:102]: Why do we need this? AFAIK it only speeds up apt install.

File: Dockerfile [hunk 3]
```diff
@@ -1,117 +1,155 @@
...
-RUN apt update && \
-	echo "Etc/UTC" > /etc/localtime && \
-	ln -s /opt/jemalloc/lib/* /usr/lib/ && \
-	apt install -y whois wget && \
-	addgroup --gid $GID mastodon && \
-	useradd -m -u $UID -g $GID -d /opt/mastodon mastodon && \
-	echo "mastodon:`head /dev/urandom | tr -dc A-Za-z0-9 | head -c 24 | mkpasswd -s -m sha-256`" | chpasswd
+RUN addgroup --gid ${GID} mastodon && \
+    useradd -m -u ${UID} -g ${GID} -d /opt/mastodon mastodon && \
+    echo "mastodon:$(head /dev/urandom | tr -dc A-Za-z0-9 | head -c 24 | mkpasswd -s -m sha-256)" | chpasswd
```
Comment (SuperSandro2000) [inline Dockerfile:121]: we could just add the account with no password.

File: Dockerfile (hunk 2 above)
Comment (shleeable) [reply Dockerfile:102]: Do you mean why do we need apt-utils?

File: Dockerfile (hunk 1 above)
Comment (shleeable) [reply Dockerfile:35]: As I mentioned, this code review is just for readability not functionality.
As this is a container, I believe it'll be /root/ but we can test.

File: Dockerfile (hunk 3 above)
Comment (shleeable) [reply Dockerfile:121]: I'm really not sure why that exists... but again. this code review is just for readability.

If it ain't broke.. leave it for another PR.

**Pull request closed without merging**